### PR TITLE
Fix/fix the account selector UI element2

### PR DESCRIPTION
**Pull request by MiroBenicio**
@mike-audi 

just for you look what I did until here, and give me any suggestions to improve what I'm doing.

### CODE REVIEW

**Comment by MiroBenicio** [comment]
@mike-audi @ricardobrg 

I've used the vue-select to build the select with the dropdown specification.

I can't manage to put the icon in the selected field yet. the docs/github issues/stackoverflow didn't help at all.

I'm gonna open a new issue for the icon selected field.

File: src/components/account/account-dropdown.vue [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+.account-select .vs__dropdown-menu {
+  border-radius: 0.5em;
+  padding: 5px 8px;
+}
+.account-select .vs__open-indicator {
+  padding: 10px 30px 10px 10px;
+  -moz-appearance: none;
+  -webkit-appearance: none;
+  appearance: none;
+  background-image: url("data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMjUiIGhlaWdodD0iMTUiIHZpZXdCb3g9IjAgMCAyNSAxNSIgZmlsbD0ibm9uZSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIj4KPHBhdGggZD0iTTEuMTgxODIgMi4wMTMxOEwxMi4yMzY0IDEyLjAxMzJMMjMgMi4wMTMxOCIgc3Ryb2tlPSJibGFjayIgc3Ryb2tlLW9wYWNpdHk9IjAuNiIgc3Ryb2tlLXdpZHRoPSIzIi8+Cjwvc3ZnPgo=");
```
Comment (mike-audi) [inline src/components/account/account-dropdown.vue:50]: Use SVG assets in the project, not remote URLs

File: src/components/account/account-dropdown.vue [hunk 2]
```diff
@@ -0,0 +1,82 @@
+<!--
+  - Copyright (c) TIKI Inc.
+  - MIT license. See LICENSE file in root directory.
+  -->
+
+<script setup lang="ts">
+import { ref, watch } from "vue";
+import { icon } from "@/service/receipt/receipt-account-type";
+const emits = defineEmits(["input"]);
```
Comment (mike-audi) [inline src/components/account/account-dropdown.vue:9]: This is not a good name for an emitter. It implies an input being entered.

File: src/components/account/account-dropdown.vue [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+const props = defineProps({
+  accounts: {
```
Comment (mike-audi) [inline src/components/account/account-dropdown.vue:11]: Why is the entire account passed in? This should simply pass back and forth the selected value (either a string or enum) and it should be set up to use a v-model

File: src/components/account/account-dropdown.vue [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+});
+</script>
+
+<template>
+  <v-select
+    :options="accounts"
+    class="account-select"
+    v-model="selectedAccount"
+  >
+    <template #option="option" v-slot:item="option">
```
Comment (mike-audi) [inline src/components/account/account-dropdown.vue:30]: Avoid putting templates inside of templates - create another file.

File: src/components/account/account-form.vue [hunk 5]
```diff
@@ -42,28 +45,32 @@ watch(
...
 );
 
 const errorMessage = ref(props.error);
 watch(
   () => props.error,
-  (newValue) => (errorMessage.value = newValue),
+  (newValue) => (errorMessage.value = newValue)
 );
+
+const teste = (accountSelected: string) => {
```
Comment (mike-audi) [inline src/components/account/account-form.vue:58]: not sure what this is, but it seems to indicate its for testing only.

File: src/components/account/account-form.vue [hunk 6]
```diff
@@ -42,28 +45,32 @@ watch(
...
-      >
-        {{ account }}
-      </option>
-    </select>
+    <account-dropdown
+      :accounts="Object.values(ReceiptAccountType)"
+      id="accounts"
+      :account="account"
+      required
+      v-model="account"
```
Comment (mike-audi) [inline src/components/account/account-form.vue:71]: there should not need to be a v-model AND an emitter

**Review by mike-audi — CHANGES_REQUESTED** [review]
see comments.

File: src/components/account/account-dropdown.vue (hunk 1 above)
Comment (MiroBenicio) [reply src/components/account/account-dropdown.vue:50]: ok! fixed!

**Comment by MiroBenicio** [comment]
@mike-audi 
Fixed most part of your comments. but I cannot figure out how to put the template slot of the v-select library in another file, because doing that I couldn't use the option slot of the component.
I'm not sure if the template have to be a direct children of the v-select
Anyway, We need to change the entire design of the select, as you said in #66

**Comment by ricardobrg** [comment]
outdated. Check #66

**Pull request closed without merging**